src/java/org/apache/cassandra/index/sai/utils/PrimaryKeyWithByteComparable.java

```diff
@@ -63,6 +63,9 @@ public int compareTo(PrimaryKey o)
         if (!(o instanceof PrimaryKeyWithByteComparable))
             throw new IllegalArgumentException("Cannot compare PrimaryKeyWithByteComparable with " + o.getClass().getSimpleName());
 
-        return ByteComparable.compare(byteComparable, ((PrimaryKeyWithByteComparable) o).byteComparable, TypeUtil.BYTE_COMPARABLE_VERSION);
+        var cmp = ByteComparable.compare(byteComparable, ((PrimaryKeyWithByteComparable) o).byteComparable, TypeUtil.BYTE_COMPARABLE_VERSION);
+        if (cmp != 0)
+            return cmp;
+        return compareSourceTables((PrimaryKeyWithByteComparable) o);
```

Comment:
Can you a bit of explanation why we want to do that?

In the issue you wrote:
> This is a minor optimization, but one that might decrease our need to resume search on a vector index in some cases.

Can you explain how that is related?
Is it that the stuff in the newest stables has higher probability to be "correct" (not shadowed) so we can skip some work processing older stuff? Just curious.